examples/PoCTutorial/src/esdl_generator.py

```diff
@@ -331,187 +342,183 @@ def create_cost_value_profile(cost_value, unit, per_unit=None, per_multiplier=No
+    cost_info = heat_source_2.costInformation
+    cost_info.investmentCosts = create_cost_value_profile(
         source_2_inv_cost, "EURO", per_unit="WATT", per_multiplier="MEGA"
     )
-    costInf.installationCosts = create_cost_value_profile(source_2_inst_cost, "EURO")
-    costInf.variableOperationalCosts = create_cost_value_profile(
+    cost_info.installationCosts = create_cost_value_profile(source_2_inst_cost, "EURO")
+    cost_info.variableOperationalCosts = create_cost_value_profile(
         source_2_varop_cost, "EURO", per_unit="WATTHOUR", per_multiplier="MEGA"
     )
-    costInf.fixedOperationalCosts = create_cost_value_profile(
+    cost_info.fixedOperationalCosts = create_cost_value_profile(
         source_2_fixop_cost, "EURO", per_unit="WATT", per_multiplier="MEGA"
     )
 
-    in_carrier = esh.get_by_id_slow(id_ret_carr)
-    out_carrier = esh.get_by_id_slow(id_primary_carr)
+    in_carrier = heat_commodity_ret
+    out_carrier = heat_commodity_primary
     create_in_out_ports(heat_source_1, in_carrier, out_carrier)
-    in_carrier = esh.get_by_id_slow(id_ret_carr)
-    out_carrier = esh.get_by_id_slow(id_primary_carr)
+    in_carrier = heat_commodity_ret
+    out_carrier = heat_commodity_primary
     create_in_out_ports(heat_source_2, in_carrier, out_carrier)
 
-    ##### Main netowrk joints joints
+    # Main netowrk joints joints
     joint_1 = esdl.Joint(id=str(uuid4()), name="Joint_1")
-    in_carrier = esh.get_by_id_slow(id_primary_carr)
-    out_carrier = esh.get_by_id_slow(id_primary_carr)
+    in_carrier = heat_commodity_primary
+    out_carrier = heat_commodity_primary
     create_in_out_ports(joint_1, in_carrier, out_carrier)
     add_geometry_point(joint_1, joint_1_coord)
     main_area.asset.append(joint_1)
 
     joint_2 = esdl.Joint(id=str(uuid4()), name="Joint_2")
-    in_carrier = esh.get_by_id_slow(id_primary_carr)
-    out_carrier = esh.get_by_id_slow(id_primary_carr)
+    in_carrier = heat_commodity_primary
+    out_carrier = heat_commodity_primary
     create_in_out_ports(joint_2, in_carrier, out_carrier)
     add_geometry_point(joint_2, joint_2_coord)
     main_area.asset.append(joint_2)
 
-    ##### Primary network pipes
-    pipe_d = esdl.PipeDiameterEnum.from_string("DN400")  # Pipe diameters
+    # Primary network pipes
+    pipe_d = esdl.PipeDiameterEnum.from_string(pipe_diam)  # Pipe diameters
 
     pipe_1 = esdl.Pipe(id=str(uuid4()), diameter=pipe_d, name="Pipe_1", state="OPTIONAL")
-    in_carrier = esh.get_by_id_slow(id_primary_carr)
-    out_carrier = esh.get_by_id_slow(id_primary_carr)
+    in_carrier = heat_commodity_primary
+    out_carrier = heat_commodity_primary
     create_in_out_ports(pipe_1, in_carrier, out_carrier)
     pipe_from = heat_source_1
     pipe_to = joint_1
-    define_pipe(pipe_1, pipe_from, pipe_to, elbow_point=pipe_1_elbow_coord)
+    connect_pipe(pipe_1, pipe_from, pipe_to, elbow_point=pipe_1_elbow_coord)
     main_area.asset.append(pipe_1)
```

Comment:
A general tip:
These lines of code are repeated several times, just with different input (e.g. pipe_name, pipe_state, carrier, from_asset, to_asset, elbow) so this could have been in one function, named create_pipe(), with these six inputs and thereby simplify these 8 lines into 1. Ofcourse same holds for the joint creation.

Reply:
I agree, it could be simplified. I updated the code. Now the pipes and joints are fully created in two respective functions.